common/tdx/quote.c

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/attestation/tdx/evidence.h>
+#include <openenclave/bits/tdx/tdxquote.h>
+#include <openenclave/internal/raise.h>
+
+#include "../common.h"
+#include "quote.h"
+
+#ifdef OE_BUILD_ENCLAVE
+#include "../../enclave/sgx/tdx_verifier.h"
+#else
+#include "../../host/tdx/quote.h"
+#endif
+
+// Max length of SGX DCAP QVL/QvE returned supplemental data
+#define MAX_SUPPLEMENTAL_DATA_SIZE 1000
+
+static const oe_uuid_t _ecdsa_uuid = {OE_FORMAT_UUID_TDX_QUOTE_ECDSA};
+
+oe_result_t oe_verify_quote_with_tdx_endorsements(
+    const uint8_t* quote,
+    size_t quote_size,
+    const uint8_t* endorsements,
+    size_t endorsements_size,
+    oe_datetime_t* input_validation_time,
+    uint32_t* verification_result,
+    uint8_t** supplemental_data,
+    size_t* supplemental_data_size)
+{
+    oe_result_t result = OE_UNEXPECTED;
+
+    uint32_t collateral_expiration_status;
+    uint32_t quote_verification_result;
+    uint8_t supplemental_data_out[MAX_SUPPLEMENTAL_DATA_SIZE] = {0};
+    uint32_t supplemental_data_size_out = 0;
+    oe_datetime_t validation_time = {0};
+    time_t expiration_check_date = 0;
+
+    if (input_validation_time)
+    {
+        validation_time = *input_validation_time;
+    }
+    else
+    {
+        // Use untrusted time if we cannot get time from inputs
+        OE_CHECK(oe_datetime_now(&validation_time));
+    }
+
+    oe_datetime_log("Validation datetime: ", &validation_time);
+
+    // Convert validation time to time_t
+    OE_CHECK(oe_datetime_to_time_t(&validation_time, &expiration_check_date));
+
+    // Call SGX DCAP QVL/QvE to verify quote
+    OE_CHECK(tdx_verify_quote(
+        &_ecdsa_uuid,
+        NULL,
+        0,
+        quote,
+        (uint32_t)quote_size,
```

Comment:
Seems best to check it fits, or even expose it in the interface rather than size_t.

Reply:
Ah, I'd missed line 42, sorry :)